lib/Doctrine/MongoDB/Aggregation/Builder.php

```diff
@@ -115,6 +115,22 @@ public function bucketAuto()
     }
 
     /**
+     * Returns statistics regarding a collection or view.
+     *
+     * $collStats must be the first stage in an aggregation pipeline, or else
```

Comment:
I'm trying to leave out as much of this server specific logic as possible. That's why I haven't added it for `$geoNear` which has the same restriction. (see https://github.com/doctrine/mongodb/pull/292#discussion_r122291605)